File: src/bot/commands/ping.ts
import { CommandInteraction, SlashCommandBuilder } from '@npm/discord.js';
import { Command } from './index.ts';

export default class Ping implements Command {
  public command(): SlashCommandBuilder {
    return new SlashCommandBuilder()
      .setName('ping')
      .setDescription('Replies with Pong!');
  }

  public async execute(interaction: CommandInteraction) {
    await interaction.reply('Pong!');
  }
}

File: src/bot/commands/ping.rs
use serenity::{
    all::{CommandInteraction, ResolvedOption},
    builder::{CreateCommand, CreateInteractionResponse, CreateInteractionResponseMessage},
    client::Context,
};

use crate::bot::BouncerState;

use super::BouncerCommand;

pub struct Command;
impl BouncerCommand for Command {
    fn command() -> CreateCommand<'static> {
        CreateCommand::new("ping").description("Pong!")
    }

    async fn execute(
        context: &Context,
        interaction: &CommandInteraction,
        _state: &BouncerState,
        _options: &[ResolvedOption<'_>],
    ) -> eyre::Result<()> {
        interaction
            .create_response(
                &context.http,
                CreateInteractionResponse::Message(
                    CreateInteractionResponseMessage::new().content("Pong!"),
                ),
            )
            .await?;

        Ok(())
    }
}
